bundler/lib/bundler/endpoint_specification.rb

```diff
@@ -31,6 +32,10 @@ def fetch_platform
       @platform
     end
 
+    def dependencies
+      @dependencies ||= @unbuilt_dependencies.map {|dep, reqs| build_dependency(dep, reqs) }
+    end
```

Comment:
You'd need to make the previous `attr_accessor` for `dependencies` just an `attr_writer`, but other than that it seems like this would work!